src/tbp/monty/frameworks/environments/embodied_data.py

```diff
@@ -138,14 +138,14 @@ def __init__(self, dataset: EnvironmentDataset, motor_system: MotorSystem, rng):
         self.dataset = dataset
         self.motor_system = motor_system
         self.rng = rng
-        self._observation, self.motor_system._policy.state = self.dataset.reset()
+        self._observation, self.motor_system._state = self.dataset.reset()
         self._action = None
         self._amount = None
         self._counter = 0
```

Comment:
This is probably outside the scope of this PR, but what is `_counter` counting? It really needs a better name.

Reply:
It is a hack to distinguish the first from the rest. Some examples:
```python
    def __next__(self):
        if self._counter == 0:
            # Return first observation after 'reset' before any action is applied
            self._counter += 1
            return self._observation
        else:
            action = self.motor_system()
            self._action = action
            self._observation, self.motor_system._state = self.dataset[action]
            self._counter += 1
            return self._observation
```
```python
    def __next__(self):
        if self._counter == 0:
            return self.first_step()
        ...
```